dialogue-core/src/main/java/com/palantir/dialogue/core/ConjureWindowedLimit.java

```diff
@@ -0,0 +1,100 @@
+/*
+ * (c) Copyright 2020 Palantir Technologies Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.dialogue.core;
+
+import com.netflix.concurrency.limits.Limit;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.atomic.AtomicReference;
+import java.util.function.Consumer;
+
+/**
+ * Changes made from {@link com.netflix.concurrency.limits.limit.WindowedLimit}:
+ *
+ * <p>1. Modify to reduce window size whenever a dropped sample is seen, rather than awaiting the whole window. 2.
+ * Change package and make package private. 3. Code style. 4. Inlined constants.
+ */
+@SuppressWarnings("NullAway") // TODO(dfox): deal with this
+class ConjureWindowedLimit implements Limit {
+    private static final long MIN_WINDOW_TIME = TimeUnit.SECONDS.toNanos(1);
+    private static final long MAX_WINDOW_TIME = TimeUnit.SECONDS.toNanos(1);
+    private static final long MIN_RTT_THRESHOLD = TimeUnit.MICROSECONDS.toNanos(100);
```

Comment:
Curious why we want this check. If we fail quickly (new hardware that can fire requests twice as fast?) we may end up ignoring all success but not failures (or the other way around).